### ISSUE TITLE
java.lang.NoSuchMethodError: lmcoursier.definitions with sbt 1.3

### ISSUE THREAD

**Issue by sbrunk** [almond-sh/almond#446]
Since we've switched to sbt 1.3.2, I'm getting this error. Might be related to https://github.com/sbt/sbt/issues/5040

```
sbt:almond> compile
[error] stack trace is suppressed; run last protocol / coursierResolutions for the full output
[error] stack trace is suppressed; run last channels / coursierResolutions for the full output
[error] stack trace is suppressed; run last logger / coursierResolutions for the full output
[error] stack trace is suppressed; run last interpreter-api / coursierResolutions for the full output
[error] stack trace is suppressed; run last scala-interpreter / coursierResolutions for the full output
[error] stack trace is suppressed; run last kernel / coursierResolutions for the full output
[error] stack trace is suppressed; run last interpreter / coursierResolutions for the full output
[error] stack trace is suppressed; run last almond-rx / coursierResolutions for the full output
[error] stack trace is suppressed; run last scala-kernel-api / coursierResolutions for the full output
[error] stack trace is suppressed; run last jupyter-api / coursierResolutions for the full output
[error] stack trace is suppressed; run last almond-spark / coursierResolutions for the full output
[error] stack trace is suppressed; run last test / coursierResolutions for the full output
[error] stack trace is suppressed; run last echo / coursierResolutions for the full output
[error] stack trace is suppressed; run last scala-kernel / coursierResolutions for the full output
[error] stack trace is suppressed; run last coursierResolutions for the full output
[error] (protocol / coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] (channels / coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] (logger / coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] (interpreter-api / coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] (scala-interpreter / coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] (kernel / coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] (interpreter / coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] (almond-rx / coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] (scala-kernel-api / coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] (jupyter-api / coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] (almond-spark / coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] (test / coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] (echo / coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] (scala-kernel / coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] (coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
```

**Comment by lucag** [comment]
If it can be of any help, a working temporary solution is to change sbt version to 1.2.8.

**Comment by alexarchambault** [comment]
The [coursier-based sbt launcher](https://github.com/coursier/sbt-launcher) is required for now. The most straightforward way of using it is by using the custom [sbt-extras script](https://raw.githubusercontent.com/coursier/sbt-extras/master/sbt). I committed it as `sbt` at the root of the almond sources.

This is required as the build is relying on some stuff from sbt-coursier, and sbt-coursier conflicts with sbt 1.3 ’s own coursier stuff.

It may be possible to get rid of sbt-coursier in the almond build, that should just require rewriting [some stuff](https://github.com/almond-sh/almond/blob/master/project/Settings.scala#L143) in `project/Settings.scala`.

**Comment by sbrunk** [comment]
Ah thanks, with the coursier based launcher it works fine. I suggest to leave this open though until we either get rid of sbt-coursier or have documented that you have to use the custom launcher.

**Comment by james64** [comment]
Hello, I would like to investigate almond code base. Simple ide functionality (jump-to-definition, find-usages) is very useful here. But I am unable to open this project in intellij due to this issue. Custom launcher run from command line works fine. I was trying to make intellij use custom runner but without success...

Do you know a way how to open it in intellij? Or what ide do you use? Thanks!

**Comment by sbrunk** [comment]
@james64 Switching temporarily to sbt 1.2.8 as suggested above by @lucag should help getting it working in intellij again.

Set `sbt.version=1.2.8` in `project/build.sbt`

**Comment by joprice** [comment]
I get this on any project using an sbt version higher than 1.2.8. I had to pin all projects to this older version. Is there any other way to use 1.3.x besides using this custom sbt runner?

**Comment by sbrunk** [comment]
Fixed by #563
